core/controllers/editor.py

```diff
@@ -157,28 +155,27 @@ def get(self, exploration_id):
             'DEFAULT_TWITTER_SHARE_MESSAGE_EDITOR': (
                 DEFAULT_TWITTER_SHARE_MESSAGE_EDITOR.value),
             'additional_angular_modules': additional_angular_modules,
-            'can_delete': rights_manager.Actor(
-                self.user_id).can_delete(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_edit': can_edit,
-            'can_modify_roles': rights_manager.Actor(
-                self.user_id).can_modify_roles(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_publicize': rights_manager.Actor(
-                self.user_id).can_publicize(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_publish': rights_manager.Actor(
-                self.user_id).can_publish(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_release_ownership': rights_manager.Actor(
-                self.user_id).can_release_ownership(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_unpublicize': rights_manager.Actor(
-                self.user_id).can_unpublicize(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_unpublish': rights_manager.Actor(
-                self.user_id).can_unpublish(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
+            'can_delete': rights_manager.check_can_delete_exploration(
+                self.user_id, self.actions, exploration_rights),
+            'can_edit': rights_manager.check_can_edit_activity(
+                self.user_id, self.actions, feconf.ACTIVITY_TYPE_EXPLORATION,
+                exploration_rights),
+            'can_modify_roles': (
+                rights_manager.check_can_modify_exploration_roles(
+                    self.user_id, self.actions, exploration_rights)),
+            'can_publicize': rights_manager.check_can_publicize_exploration(
```

Comment:
Oh, a further note btw: the above comment doesn't apply to the publish/unpublish stuff, which should still be fixed in the originally agreed-upon way.